pkg/loadbalancer/proxy.go

```diff
@@ -43,98 +68,113 @@ type endpoint struct {
-        '@type': type.googleapis.com/envoy.extensions.filters.udp.udp_proxy.v3.UdpProxyConfig
-        stat_prefix: cluster_{{$index}}
-        matcher:
-          on_no_match:
-            action:
-              name: route
-              typed_config:
-                '@type': type.googleapis.com/envoy.extensions.filters.udp.udp_proxy.v3.Route
-                cluster: cluster_{{$index}}
+        "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
+        access_log:
+        - name: envoy.file_access_log
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.access_loggers.stream.v3.StdoutAccessLog
+        stat_prefix: tcp_proxy
+        cluster: cluster_{{$index}}
         {{- if eq $.SessionAffinity "ClientIP"}}
-        hash_policies:
-          source_ip: true
+        hash_policy:
+          source_ip: {}
         {{- end}}
-        upstream_socket_config:
-          max_rx_datagram_size: 9000
-    {{- else }}
-    filter_chains:
-      - filters:
-        - name: envoy.filters.network.tcp_proxy
-          typed_config:
-            "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
-            stat_prefix: destination
-            cluster: cluster_{{$index}}
-            {{- if eq $.SessionAffinity "ClientIP"}}
-            hash_policy:
-              source_ip: {}
-            {{- end}}
-    {{- end}}
-  {{- end }}
-
-  clusters:
-  {{- range $index, $servicePort := .ServicePorts }}
-  - name: cluster_{{$index}}
-    connect_timeout: 5s
-    type: STATIC
-    {{- if eq $.SessionAffinity "ClientIP"}}
-    lb_policy: RING_HASH
-    {{- else}}
-    lb_policy: RANDOM
-    {{- end}}
-    health_checks:
-      - timeout: 5s
-        interval: 3s
-        unhealthy_threshold: 3
-        healthy_threshold: 1
-        always_log_health_check_failures: true
-        always_log_health_check_success: true
-        http_health_check:
-          path: /healthz
-    load_assignment:
-      cluster_name: cluster_{{$index}}
-      endpoints:
-      {{- range $address := $servicePort.Cluster }}
-        - lb_endpoints:
-          - endpoint:
-              health_check_config:
-                port_value: {{ $.HealthCheckPort  }}
+  {{- end}}
+{{- end }}
+`
+
+// proxyCDSConfigTemplate is the loadbalancer config template for clusters
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/health_check.proto#envoy-v3-api-msg-config-core-v3-healthcheck-httphealthcheck
+const proxyCDSConfigTemplate = `
+resources:
+{{- range $index, $servicePort := .ServicePorts }}
+- "@type": type.googleapis.com/envoy.config.cluster.v3.Cluster
+  name: cluster_{{$index}}
+  connect_timeout: 5s
+  type: STATIC
+  {{- if eq $.SessionAffinity "ClientIP"}}
+  lb_policy: RING_HASH
+  {{- else}}
+  lb_policy: RANDOM
+  {{- end}}
+  health_checks:
+  - timeout: 5s
+    interval: 3s
+    unhealthy_threshold: 2
+    healthy_threshold: 2
+    no_traffic_interval: 5s
```

Comment:
this was the problem 

```
no_traffic_interval
([Duration](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#duration)) The “no traffic interval” is a special health check interval that is used when a cluster has never had traffic routed to it. This lower interval allows cluster information to be kept up to date, without sending a potentially large amount of active health checking traffic for no reason. Once a cluster has been used for traffic routing, Envoy will shift back to using the standard health check interval that is defined. Note that this interval takes precedence over any other.

The default value for “no traffic interval” is 60 seconds.
```

https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/health_check.proto#envoy-v3-api-msg-config-core-v3-healthcheck-httphealthcheck

by default it wait 60 seconds and sends traffic to nodes failing the healtcheck, because the second healthchec was not done after that ... that is my observation, there should be a better explanation